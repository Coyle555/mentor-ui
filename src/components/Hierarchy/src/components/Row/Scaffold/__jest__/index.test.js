import React from 'react';
import renderer from 'react-test-renderer';
import { Scaffold } from '../index';

test('Scaffold of a node at root', () => {
	const tree = [{ hasSibling: false }];
	const render = renderer.create(<Scaffold level={0} nodeIndex={0} tree={tree} />).toJSON();

	expect(render).toMatchSnapshot();
});

test('Scaffold of a node with a parent', () => {
	const tree = [{ hasSibling: false }, { hasSibling: false, parent: 0 }];
	const render = renderer.create(<Scaffold level={1} nodeIndex={1} tree={tree} />).toJSON();

	expect(render).toMatchSnapshot();
});

test('Scaffold of a node with a parent and child', () => {
	const tree = [{ hasSibling: false }, { hasSibling: false, parent: 0 }, { hasSibling: false, parent: 1 }];
	const render = renderer.create(<Scaffold level={1} nodeIndex={1} tree={tree} />).toJSON();

	expect(render).toMatchSnapshot();
});

test('Scaffold of a node with a parent and sibling', () => {
	const tree = [{ hasSibling: false }, { hasSibling: true, parent: 0 }, { hasSibling: false, parent: 0 }];
	const render = renderer.create(<Scaffold level={1} nodeIndex={1} tree={tree} />).toJSON();

	expect(render).toMatchSnapshot();
});
