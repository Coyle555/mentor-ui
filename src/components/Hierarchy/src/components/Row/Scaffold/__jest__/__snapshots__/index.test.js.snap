// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scaffold of a node at root 1`] = `
<div
  className="mui-line-block mui-line-half-horizontal-right"
/>
`;

exports[`Scaffold of a node with a parent 1`] = `
Array [
  <div
    className="mui-line-block"
  />,
  <div
    className="mui-line-block mui-line-half-horizontal-right mui-line-half-vertical-top"
  />,
]
`;

exports[`Scaffold of a node with a parent and child 1`] = `
Array [
  <div
    className="mui-line-block"
  />,
  <div
    className="mui-line-block mui-line-half-horizontal-right mui-line-half-vertical-top"
  />,
]
`;

exports[`Scaffold of a node with a parent and sibling 1`] = `
Array [
  <div
    className="mui-line-block"
  />,
  <div
    className="mui-line-block mui-line-half-horizontal-right mui-line-full-vertical"
  />,
]
`;
